src/ngraph/builder/quantization_util.hpp

```diff
@@ -99,6 +99,59 @@ namespace ngraph
                     (std::pow(2, -24) * static_cast<double>(max_abs32 / max_abs8)));
                 return scale;
             }
+
+            static inline float get_quantize_scale(const std::shared_ptr<Node> min_input,
+                                                   const std::shared_ptr<Node> max_input,
+                                                   bool is_signed)
+            {
+                auto min_input_const_op = std::static_pointer_cast<ngraph::op::Constant>(min_input);
+                auto max_input_const_op = std::static_pointer_cast<ngraph::op::Constant>(max_input);
+                float input_min_range =
+                    *(static_cast<float const*>(min_input_const_op->get_data_ptr()));
+                float input_max_range =
+                    *(static_cast<float const*>(max_input_const_op->get_data_ptr()));
+
```

Comment:
Can we use `get_vector` here (instead of get_data_ptr), assert that it is of size 1 and then read the first element into input_min_range

Reply:
trying to understand the advantage of it. Also, the plan in future is to have channel specific scales, so the mins and maxes will we vectors of length > 1

Reply:
atleast in mxnet, these are passed as vectors. I would prefer vector too. (we currently have to use extra reshape)

Reply:
Ah, I see. In TF min/max is a single number, hence was thinking that the min/max const is of size 1. If per-channel info is to be supported then no sense assert size==1

Reply:
@nbpatel can we make min/max be vectors ? above static_cast will need to change accordingly.

Reply:
so are you guys suggesting having the min and max as vectors in the builder and then converting to a ngraph constant op before calling the ngraph quantize?

Reply:
We'll need per-channel. Add a broadcast if they are scalars.